mt_with_external_memory/mt_with_external_memory.py

```diff
@@ -67,24 +54,40 @@ class ExternalMemory(object):
     :type name: basestring
     :param mem_slot_size: Size of memory slot/vector.
     :type mem_slot_size: int
-    :param boot_layer: Boot layer for initializing memory. Sequence layer
-                       with sequence length indicating the number of memory
-                       slots, and size as mem_slot_size.
+    :param boot_layer: Boot layer for initializing the external memory. The
+                       sequence layer has sequence length indicating the number
+                       of memory slots, and size as memory slot size.
     :type boot_layer: LayerOutput
     :param readonly: If true, the memory is read-only, and write function cannot
                      be called. Default is false.
     :type readonly: bool
+    :param enable_interpolation: If set true, the read/write addressing weights
+                                 will be interpolated with the weights in the
+                                 last step, with the affine coefficients being
+                                 a learnable gate function.
+    :type enable_interpolation: bool
     """
 
-    def __init__(self, name, mem_slot_size, boot_layer, readonly=False):
+    def __init__(self,
+                 name,
+                 mem_slot_size,
+                 boot_layer,
+                 readonly=False,
+                 enable_interpolation=True):
         self.name = name
         self.mem_slot_size = mem_slot_size
         self.readonly = readonly
+        self.enable_interpolation = enable_interpolation
         self.external_memory = paddle.layer.memory(
             name=self.name,
             size=self.mem_slot_size,
             is_seq=True,
             boot_layer=boot_layer)
+        # prepare a constant (zero) intializer for addressing weights 
+        self.zero_addressing_init = paddle.layer.slope_intercept(
+            input=paddle.layer.fc(input=boot_layer, size=1),
+            slope=0.0,
+            intercept=0.0)
```

Comment:
Yes, but that might be another kind of inconvenience, since I have to prepare and pass in different dummy data layer for different ExternalMemory instances (with different size). Here, I hope to create it just within ExternalMemory, and no more work outside ExternalMemory class.